tests/target/unsafe_extern_blocks.rs

```diff
@@ -0,0 +1,18 @@
+// See tracking issue for unsafe_extern_blocks
+// https://github.com/rust-lang/rust/issues/123743
+
+#![feature(unsafe_extern_blocks)]
+
+safe static TEST1: i32;
```

Comment:
I'm not sure that formatting the `safety` here is necessary per this comment https://github.com/rust-lang/rust/pull/124482/files#r1608953719. For now it parses, so I don't think there's an issue, though I'm happy to revert the changes to `StaticParts` if we don't think this is necessary.

Reply:
Let's keep it, since it doesn't hurt.